WireAnalytics/Sources/WireAnalytics/Events/Predefined/AnalyticsEvent.User.swift

```diff
@@ -0,0 +1,106 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+public extension AnalyticsEvent {
+
+    enum User {
+
+        public enum IndividualToTeamMigration {
+
+            /// The various steps of the individual to team migration flow.
+
+            public enum Step: Int {
+                case disclaimer = 1
+                case teamName
+                case confirmation
+            }
+
+            /// The actions a user can choose on the dismiss confirmation alert.
+
+            public enum CancelConfirmationAction {
+                case `continue`
+                case leave
+            }
+
+            /// The actions a user can choose on the final confirmation scren.
+
+            public enum CompletedAction {
+                case backToWire
+                case openTeamManagement
+            }
+
+        }
+
+        /// An event tracking when the user reaches a certain step of the individual to team migration flow.
+
+        public static func personalTeamCreationFlowStarted(at step: IndividualToTeamMigration.Step) -> AnalyticsEvent {
+            AnalyticsEvent(name: "user.personal-team-creation-flow-started") {
+                SegmentationEntry(key: "step_modalcreateteam", value: step.rawValue)
+            }
+        }
+
+        /// An event tracking when the user dismisses the individual to team migration flow at a certain step.
+
+        public static func personalToTeamMigrationFlowStopped(
+            at step: IndividualToTeamMigration
+                .Step
+        ) -> AnalyticsEvent {
+            AnalyticsEvent(name: "user.personal-team-creation-flow-stopped") {
+                switch step {
+                case .disclaimer:
+                    SegmentationEntry(key: "modal_disclaimers", value: true)
+                case .teamName:
+                    SegmentationEntry(key: "modal_team-name", value: true)
+                case .confirmation:
+                    SegmentationEntry(key: "modal_confirmation", value: true)
+                }
+            }
+        }
```

Comment:
Not sure why, but too late to argue about it:
For the flow-started event we send a step number, for stopped event we send different keys depending on the step.

Reply:
At the end of the day it probably won't be us who digest the analytics data so if this is what was asked then I guess so be it.